test/cppsuite/test_harness/workload/workload_tracking.h

```diff
@@ -137,19 +138,19 @@ class workload_tracking : public component {
                 sweep_key = static_cast<char *>(dstrdup(key));
                 sweep_collection_id = collection_id;
             }
-            if (timestamp <= oldest_ts) {
+            if (ts <= oldest_ts) {
                 if (globally_visible_update_found) {
                     if (_trace_level == LOG_TRACE)
                         log_msg(LOG_TRACE,
                           std::string("workload tracking: Obsoleted update, key=") + sweep_key +
-                            ", timestamp=" + std::to_string(timestamp) +
+                            ", timestamp=" + std::to_string(ts) +
                             ", oldest_timestamp=" + std::to_string(oldest_ts));
                     testutil_check(_sweep_cursor->remove(_sweep_cursor.get()));
-                } else {
+                } else if (static_cast<tracking_operation>(op_type) == tracking_operation::INSERT) {
```

Comment:
Why do you check the operation type ? What if it is DELETE_KEY ?

Reply:
In that case, I don't want to delete everything behind it since validation expects an insert to come before a delete.

Reply:
Let's say we insert a key, then update N times the same key, then finally delete it and move the oldest_ts to a time after the deletion of that key.
Does it mean we will keep the delete operation in the tracking table even though it happened before the oldest_ts ?

Reply:
To be clear the logic here will mean that we only keep one additional insert behind the delete? @tetsuo-cpp

Reply:
Yes. Basically, something isn't considered "globally visible" if it is a delete. So when we find the first non-delete globally visible update, we delete everything before it.

That's because validation expects that a delete is applied when an existing value is already there.

Reply:
Thanks the clarification @tetsuo-cpp